cpp/src/strings/utilities.cu

```diff
@@ -129,6 +129,13 @@ special_case_mapping const* get_special_case_mapping_table()
   });
 }
 
+int64_t get_offset64_threshold()
+{
+  auto const threshold  = std::getenv("LIBCUDF_LARGE_STRINGS_THRESHOLD");
+  std::size_t const rtn = threshold != nullptr ? std::atol(threshold) : 0;
+  return rtn > 0 ? rtn : std::numeric_limits<int32_t>::max();
```

Comment:
It's possible to set `LIBCUDF_LARGE_STRINGS_THRESHOLD` to be greater than `std::numeric_limits<int32_t>::max()`.
Is there a reason behind choosing return type as `int64_t`?
For the purpose of this function, is `int32_t` enough?

Reply:
It will always be compared to an int64 variable.
Is there a reason to not support int64?

Reply:
Maximum value `LIBCUDF_LARGE_STRINGS_THRESHOLD` that's valid is `std::numeric_limits<int32_t>::max()` for string column purpose. It's representable with `int32_t`. `int64_t` is not necessary, unless there is a reason besides string column 32bit/64bit offsets choice.

What happens if a user sets `LIBCUDF_LARGE_STRINGS_THRESHOLD`  > int32 max? what's the expected behaviour? We should document valid range and behaviour of this environment variable in this function documentation.